### ISSUE TITLE
Typescript complaining after update to 0.4.1

### ISSUE THREAD

**Issue by w3bdesign** [coreyward/react-portable-text#16]
I am getting the following error after updating to 0.4.1:

```
Type '{ className: string; content: [object]; serializers: { code: (props: ISerializerCode) => Element; link: (props: ISerializerLink) => Element; }; }' is missing the following properties from type 'Pick<{ dataset: string; projectId: string; content: [object]; className: string; serializers: object; }, "className" | "content" | "projectId" | "dataset" | "serializers">': projectId, dataset
```

I tested out my theory by temporarily modifying the types to this:

```
declare function PortableText({ content, className, serializers, dataset, projectId, ...additionalOptions }: {
    dataset?: string; <- No longer required (optional)
    projectId?: string; <- No longer required (optional)
    content: [object];
    className: string;
    serializers: object;
}): any;
```

It seems that dataset and projectId is required now, but nothing in the documentation states this?

**Comment by ETAnderson** [comment]
I ran into the same issue and received identical error message upon compilation. Updating the types in PortableText with additions of className and projectId like so 
`export default PortableText;
export function blockContentToPlainText(blocks?: [any]): string;
/**
 * Renders an array of Portable Text blocks as React components.
 *
 * @param {object} props
 * @param {[object]} props.content Array of portable text blocks
 * @param {string} props.className Optional className
 * @param {object} props.serializers Optional serialization overrides
 * @returns
 */
declare function PortableText({ content, className, serializers, dataset, projectId, ...additionalOptions }: {
    content: [object];
    className: string;
    serializers: object;
    dataset: string;
    projectId: string;
}): any;
declare namespace PortableText {
    namespace propTypes {
        const content: any;
        const className: any;
        const serializers: any;
        const dataset: string;
        const projectId: string;
    }
}
`
does work, but be prepared to update the types each time your package manager verifies it's modules, which has caused a deployment issue if i don't include my node_modules into the repository I deploy from.

**Comment by w3bdesign** [comment]
A temporary workaround seems to be to add empty values for projectId and dataset like this:

```
<PortableText content={content.text} dataset="" projectId="" />
```

**Comment by w3bdesign** [comment]
A temporary workaround seems to be to add empty values for projectId and dataset like this:

```
<PortableText content={content.text} dataset="" projectId="" />
```

@coreyward

**Comment by coreyward** [comment]
I just published 0.4.2 to fix the JSDoc declarations. Give that a try and see if this is any better. If not I'm not sure what more I'm able to do—the typings are being generated by TypeScript and they seem to all be optional.

**Comment by w3bdesign** [comment]
Fixed in https://github.com/coreyward/react-portable-text/pull/17

Please take a look @coreyward

**Comment by coreyward** [comment]
Fixed in bb86e19000bd3baf54f2ea97e34f4be7e6689f33 (#17).
